lib/src/drawConfigContext.dart

```diff
@@ -87,31 +126,68 @@ class DrawConfigContext {
     if (primaryFile.exists() == false) {
       primaryFile = new File(this._globalConfigPath.toString());
     }
+  }
 
-    //Parse the ini file.
-    _customConfig = new Config.fromStrings(primaryFile.readAsLinesSync());
-    _initializeAttributes();
+  void _fieldInitializer(type, params) {
+    params.forEach((value) => _initializeField(type, value));
   }
 
   /// Initialize the attributes of the configuration object using the ini file.
-  void _initializeAttributes() {
-    _shortURL = _fetchDefault('short_url');
-    checkForUpdates = _configBool(_fetchDefault('check_for_updates'));
-
-    oauthUrl = _fetch('oauth_url');
-    redditUrl = _fetch('reddit_url');
-
-    //The use of null aware operators here is to give highest precedence to
-    //passed in values to the constructor.
-    clientId ??= _fetchOrNotSet('client_id');
-    clientSecret ??= _fetchOrNotSet('client_secret');
-    httpProxy ??= _fetchOrNotSet('http_proxy');
-    httpsProxy ??= _fetchOrNotSet('https_proxy');
-    _redirectUri ??= _fetchOrNotSet('redirect_uri');
-    refreshToken ??= _fetchOrNotSet('refresh_token');
-    password ??= _fetchOrNotSet('password');
-    userAgent ??= _fetchOrNotSet('user_agent');
-    username ??= _fetchOrNotSet('username');
+  void _initializeField(String type, String param) {
+    if (type == kShortUrl) {
+      _shortURL = _fetchDefault('short_url');
+    } else if (type = kCheckForUpdates) {
+      checkForUpdates = _configBool(_fetchDefault('check_for_updates'));
+    } else if (type = kKind) {
+      //TODO(kc3454): Learn how to do this one.
+    } else if (type = kOptionalField) {
+      String value = _fetchOrNotSet(param);
+      if (value != null) {
+        switch (param) {
+          case 'cliend_id':
```

Comment:
also, just to be safe, it might be worth running spellcheck on any constant strings you add to make sure everything is correct.